### DISCUSSION TITLE
Update Microsoft Identify Provider

### DISCUSSION THREAD

**Discussion by miguoliang** [keycloak/keycloak#16406] [Ideas]
Microsoft authentication service has migrated to Azure Active Directory for months, and the most important change is the OAuth2 endpoints urls using a Tenant ID instead of the "common", for example, in the past the Authorization url is hard coded like https://login.microsoftonline.com/common/oauth2/v2.0/authorize, but now, user should retrive the url from the endpoints that listed in Azure AD like https://login.microsoftonline.com/13472399-ab64-4ebc-903b-4eca6bb88e43/oauth2/v2.0/authorize.
![image](https://user-images.githubusercontent.com/14083659/211963805-48be7c43-d72a-4468-b019-57716ccc2ac5.png)
Therefore, I wonder, if we need make some changes to make build-in Microsoft Identify Provider working again?
If yes, I am glad to make a PR, otherwise, I recommend remove the build-in Microsoft Identify Provider from next release on.
So, what is our choice?

**Comment by magicrobotmonkey** [comment]
do you know when this changed over on microsofts side?
